api/models/error.go

```diff
@@ -293,3 +298,32 @@ func NewAPIErrorWrapper(apiErr APIError, rootErr error) APIErrorWrapper {
 		root:     rootErr,
 	}
 }
+
+type RetryableError interface {
+	APIErrorWrapper() APIErrorWrapper
```

Comment:
my nit with this is that this interface does too much. this interface only needs to implement the ability to set an error as retryable. an error can then still implement APIErrorWrapper to get at a root error if it wants to in addition to being a RetryableError, but it needn't be an APIErrorWrapper itself. then we can compose the legos